core/src/main/java/org/apache/accumulo/core/security/ColumnVisibility.java

```diff
@@ -505,6 +524,31 @@ public ColumnVisibility(byte[] expression) {
     validate(expression);
   }
 
+  /**
+   * Creates a column visibility for a Mutation from an AccessExpression.
+   *
+   * @param expression visibility expression, encoded as UTF-8 bytes
+   * @see #ColumnVisibility(String)
+   * @since 2.1.4
+   */
+  public ColumnVisibility(AccessExpression expression) {
+    // AccessExpression is a validated immutable object, so no need to re validate
+    this.expression = expression.getExpression().getBytes(UTF_8);
```

Comment:
Is this just an optimization, or does calling `validate` cause an issue? A client could use AccessExpression's and then create a ColumnVisibilty using the byte[] constructor.

Reply:
This is an optimization to avoid calling `validate` for performance reasons. If an AccessExpression is passed in then it has already performed a `validate` operation previous and does not need to parse the expression.

Reply:
We could a constructor then that doesn't perform validation, mark it deprecated, and in the javadoc:

1.  that this constructor was added for clients that want to use Accumulo-Access before Accumulo 4.0
2. this will be removed in 4.0
3. using this incorrectly (with an invalid column visibility) will likely cause exceptions and no data to be returned

Reply:
I like that approach.

Reply:
Considering that we already have a `ColumnVisibility(byte[])` constructor, we would likely need to create a `ColumnVisibility(byte[], boolean)` constructor.

Reply:
> Considering that we already have a `ColumnVisibility(byte[])` constructor, we would likely need to create a `ColumnVisibility(byte[], boolean)` constructor.

I added that constructor but made it private. Then I added a new static method to use the private constructor called `fromAccessExpression` so that creation of these unvalidated ColumnVisibilities is very explicit.

```
public static ColumnVisibility fromAccessExpression(byte[] incomingExpression) {
    return new ColumnVisibility(incomingExpression, false);
  }
```

The name should probably be changed, so I'm open to suggestions.